src/Examples/Flowsheets/HDA_flowsheet_with_distillation_solution_testing.ipynb

```diff
@@ -0,0 +1,1773 @@
+    "\n",
+    "# Initialize the distillation column\n",
+    "m.fs.D101.initialize(outlvl=idaeslog.INFO)                        "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Adding expressions to compute capital and operating costs\n",
+    "\n",
+    "In this section, we will add a few Expressions that allow us to evaluate the performance. Expressions provide a convenient way of calculating certain values that are a function of the variables defined in the model. For more details on Expressions, please refer to: https://pyomo.readthedocs.io/en/latest/pyomo_modeling_components/Expressions.html"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Expression to compute the total cooling cost\n",
+    "m.fs.cooling_cost = Expression(expr=0.25e-7 * (-m.fs.F101.heat_duty[0]) +\n",
+    "                                0.2e-7 * (-m.fs.D101.condenser.heat_duty[0]))\n",
+    "\n",
+    "# Expression to compute the total heating cost\n",
+    "m.fs.heating_cost = Expression(expr=2.2e-7 * m.fs.H101.heat_duty[0] +\n",
+    "                                1.2e-7 * m.fs.H102.heat_duty[0] +\n",
+    "                                1.9e-7 * m.fs.D101.reboiler.heat_duty[0])\n",
+    "\n",
+    "# Expression to compute the total operating cost\n",
+    "m.fs.operating_cost = Expression(expr=(3600 * 24 * 365 *\n",
+    "                                (m.fs.heating_cost + m.fs.cooling_cost)))\n",
+    "\n",
+    "# Expression to compute the total capital cost\n",
+    "m.fs.capital_cost = Expression(expr=1e5*m.fs.R101.volume[0])    "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Solve the entire flowsheet"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": [
+     "testing",
+     "remove_cell"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "# Check that the degrees of freedom is zero\n",
+    "assert degrees_of_freedom(m) == 0"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "solver.solve(m, tee=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": [
+     "testing",
+     "remove_cell"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "# Check solver solve status\n",
+    "from pyomo.environ import TerminationCondition\n",
+    "assert results.solver.termination_condition == TerminationCondition.optimal"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Analyze the Results of the Square Problem\n",
+    "\n",
+    "How much is the total cost (operating_cost + capital_cost), operating_cost, capital_cost, benzene purity in the distillate from the distilation column, and conversion of toluene in the reactor?"
```

Comment:
I would remove the underscores from the names here - you are describing the quantities rather than using the actual variable names,.